SI-9097 Remove spurious warning about conflicting filenames

lrytz opened the pull request:
When using delambdafy:method, closure classes are generated late.
The class is added to a map and integrated into the PackageDef in
transformStats.

When declaring a package object, there are potentially multiple
PackageDefs for the same package. In this case, the closure class
was added to all of them. As a result, GenASM / GenBCode would run
multiple times on the closure class. In GenBCode this would trigger
a warning about conflicting filenames.

lrytz committed:
SI-9097 Remove spurious warning about conflicting filenames

When using delambdafy:method, closure classes are generated late.
The class is added to a map and integrated into the PackageDef in
transformStats.

When declaring a package object, there are potentially multiple
PackageDefs for the same package. In this case, the closure class
was added to all of them. As a result, GenASM / GenBCode would run
multiple times on the closure class. In GenBCode this would trigger
a warning about conflicting filenames.

lrytz commented:
The spurious warning can be seen here: https://scala-webapps.epfl.ch/jenkins/view/2.11.x/job/scala-nightly-genbcode-2.11.x/263/consoleFull

```
[quick.library] /localhome/jenkins/b/workspace/scala-nightly-genbcode-2.11.x/src/library/scala/collection/parallel/immutable/package.scala:33: warning: Class scala.collection.parallel.immutable.Repetition$ParIterator$lambda$$psplit$2 differs only in case from scala.collection.parallel.immutable.Repetition$ParIterator$lambda$$psplit$2. Such classes will overwrite one another on case-insensitive filesystems.
[quick.library]         for ((start, end) <- incr.init zip incr.tail) yield new ParIterator(i + start, (i + end) min until, elem)
[quick.library]                           ^
```

lrytz commented:
Added a more robust test. Review by @retronym.

som-snytt committed:
SI-9097 Consolidate test

`pos` test is subsumed by `run`.

som-snytt commented:
It's worth noting that the message "differs in case" is imprecise. Or maybe not worth noting.

Here's a consolidation of the tests, just to see if I could:

https://github.com/lrytz/scala/pull/2

lrytz commented:
@som-snytt: thanks!

retronym committed:
Refactor transformStats impls for consistency

Uses the same idiom in `Flatten` and `LambdaLift` as is established
in `Extender` and (recently) `Delambdafy`. This avoids any possibility
of adding a member to a package twice, as used to happen in SI-9097.

retronym commented:
I took a look at other `transformStats` implementations to see if any kindred bugs lie in wait. This patch aims for consistent use of the `map remove currentOwner` idiom used here and in `Extender`. https://github.com/lrytz/scala/pull/3

retronym commented:
LGTM in any case.

lrytz commented:
PLS REBUILD/pr-scala@486f92c5ddd6f02cdcd6e32329ce92f90a9fa1c9

scala-jenkins commented:
(kitty-note-to-self: ignore 72714848)
:cat: Roger! Rebuilding pr-scala for 486f92c5. :rotating_light:
